src/StdChains.sol

```diff
@@ -0,0 +1,103 @@
+    Vm private constant vm = Vm(address(uint160(uint256(keccak256("hevm cheat code")))));
+
+    struct Chain {
+        // The chain name, using underscores as the separator to match `foundry.toml` conventions.
+        string name;
+        // The chain's Chain ID.
+        uint256 chainId;
+        // A default RPC endpoint for this chain.
+        // NOTE: This default RPC URL is included for convenience to facilitate quick tests and
+        // experimentation. Do not use this RPC URL for production test suites, CI, or other heavy
+        // usage as you will be throttled and this is a disservice to others who need this endpoint.
+        string rpcUrl;
+    }
+
+    struct Chains {
+        Chain Anvil;
+        Chain Hardhat;
+        Chain Mainnet;
+        Chain Goerli;
+        Chain Sepolia;
+        Chain Optimism;
+        Chain OptimismGoerli;
+        Chain ArbitrumOne;
+        Chain ArbitrumOneGoerli;
+        Chain ArbitrumNova;
+        Chain Polygon;
+        Chain PolygonMumbai;
+        Chain Avalanche;
+        Chain AvalancheFuji;
+        Chain BnbSmartChain;
+        Chain BnbSmartChainTestnet;
+        Chain GnosisChain;
+    }
+
+    Chains stdChains = Chains({
+        Anvil: Chain("Anvil", 31337, "http://127.0.0.1:8545"),
+        Hardhat: Chain("Hardhat", 31337, "http://127.0.0.1:8545"),
+        Mainnet: Chain("Mainnet", 1, "https://api.mycryptoapi.com/eth"),
+        Goerli: Chain("Goerli", 5, "https://goerli.infura.io/v3/84842078b09946638c03157f83405213"), // Default Infura key from ethers.js: https://github.com/ethers-io/ethers.js/blob/c80fcddf50a9023486e9f9acb1848aba4c19f7b6/packages/providers/src.ts/infura-provider.ts
+        Sepolia: Chain("Sepolia", 11155111, "https://rpc.sepolia.dev"),
+        Optimism: Chain("Optimism", 10, "https://mainnet.optimism.io"),
+        OptimismGoerli: Chain("OptimismGoerli", 420, "https://goerli.optimism.io"),
+        ArbitrumOne: Chain("ArbitrumOne", 42161, "https://arb1.arbitrum.io/rpc"),
+        ArbitrumOneGoerli: Chain("ArbitrumOneGoerli", 421613, "https://goerli-rollup.arbitrum.io/rpc"),
+        ArbitrumNova: Chain("ArbitrumNova", 42170, "https://nova.arbitrum.io/rpc"),
+        Polygon: Chain("Polygon", 137, "https://polygon-rpc.com"),
+        PolygonMumbai: Chain("PolygonMumbai", 80001, "https://rpc-mumbai.matic.today"),
+        Avalanche: Chain("Avalanche", 43114, "https://api.avax.network/ext/bc/C/rpc"),
+        AvalancheFuji: Chain("AvalancheFuji", 43113, "https://api.avax-test.network/ext/bc/C/rpc"),
+        BnbSmartChain: Chain("BnbSmartChain", 56, "https://bsc-dataseed1.binance.org"),
+        BnbSmartChainTestnet: Chain("BnbSmartChainTestnet", 97, "https://data-seed-prebsc-1-s1.binance.org:8545"),
+        GnosisChain: Chain("GnosisChain", 100, "https://rpc.gnosischain.com")
+    });
+
+    // TODO how can we hide the compiler warnings by default? We can't remove the visibility since it's needed for ^0.6.2 compatibility.
+    constructor() internal {
```

Comment:
Create a `private` or `internal` function `_constructor` that returns a placeholder value, and move the logic there. Then, add a helper `private constant` variable and assign the value to it by calling `_constructor`.

```
bool private constant constructorTrigger = _constructor();
```

Reply:
Ah man this is super clever and I was excited about it. Unfortunately I don't think it'll work though:

```
Compiler run failed
error[8349]: TypeError: Initial value for constant variable has to be compile-time constant.
  --> src/StdChains.sol:18:44:
   |
18 |     uint256 private constant CONSTRUCTOR = _constructor();
````

And we can't use immutables since it's not supported in 0.6.2. @mattsse is there a way forge can hide warnings based on the contract they come from so we can suppress logging of `StdChains` constructor warnings

Reply:
@mds1, is it because we marked it `constant`? It doesn’t have to be; just thought it’d look more formal that way.

Reply:
Oh right, of course it doesn't need to be constant 🤦

Resolved in 14e1d03

Reply:
Huh actually that causes the initialization test to fail as you can see in the CI for that commit, so for some reason this approach actually does not work

Reply:
*Moved the comment from the wrong thread:*

It's because `stdChains` is initialized after our `_constructor` is run, so it overrides the setup.

Solution: move `stdChains` initialization into our `_constructor`, before setting the values.